autoload/sj/elm.vim

```diff
@@ -0,0 +1,247 @@
+
+  if from[0] == 0
+    call sj#PopCursor()
+
+    return [-1, -1]
+  end
+
+  call cursor(from[0], from[1])
+  normal! %
+
+  if line('.') != currentLine
+    call sj#PopCursor()
+
+    return [-1, -1]
+  endif
+
+  let to = col('.')
+
+  call sj#PopCursor()
+
+  return [from[1], to]
+endfunction
+
+" function! sj#elm#CurrentLineOutermostBraces(column) {{{2
+"
+" Returns the colums corresponding to the bracing pair furthest
+" to and surrounding given column as a list ([opening, closing]).
+" Returns [-1, -1] when there is none on the same line.
+function! sj#elm#CurrentLineOutermostBraces(column)
+  if a:column < 1
+    return [-1, -1]
+  endif
+
+  let currentMatch = sj#elm#CurrentLineClosestBraces(a:column)
+
+  if currentMatch[0] < 1
+    return [-1, -1]
+  endif
+
+  let betterMatch = sj#elm#CurrentLineOutermostBraces(currentMatch[0] - 1)
+
+  if betterMatch[0] < 1
+    return currentMatch
+  endif
+
+  return betterMatch
+endfunction
+
+" function! sj#elm#ClosestMultilineBraces() {{{2
+"
+" Returns the position of the closest pair of multiline braces
+" around the cursor.
+"
+" The positions are given as a couple of arrays:
+"
+"   [[startLine, startCol], [endLine, endCol]]
+"
+" When no position is found, returns:
+"
+"   [[-1, -1], [-1, -1]]
+function! sj#elm#ClosestMultilineBraces()
+  call sj#PushCursor()
+
+  let skip = sj#elm#SkipSyntax()
+
+  let currentLine = line('.')
+
+  normal! $
+
+  let [toLine, toCol] = searchpairpos('[{(\[]', '', '[})\]]', '', skip, line('$'))
+
+  if toLine < currentLine
+    call sj#PopCursor()
+    return [[-1, -1], [-1, -1]]
+  endif
+
+  normal! %
+
+  let fromLine = line('.')
+  let fromCol = col('.')
+
+  if fromLine >= toLine
+    call sj#PopCursor()
+    return [[-1, -1], [-1, -1]]
+  endif
+
+  call sj#PopCursor()
+
+  return [[fromLine, fromCol], [toLine, toCol]]
+endfunction
+
+" function! sj#elm#JoinOuterBraces(text)
+"
+" Removes spaces separating the first and last char of a string
+" with the closest non-space char found.
+"
+" ex:
+"
+"   [ 1, 2, 3 ] => [1, 2, 3]
+function! sj#elm#JoinOuterBraces(text)
+  return substitute(substitute(a:text, '\s*\(.\)$', '\1', ''), '^\(.\)\s*', '\1', '')
+endfunction
+
+" function! sj#elm#JoinLines(text)
+"
+" Joins lines in text, removing complementary spaces around
+" newline chars when the last line starts with a ',' and keeping
+" one whitespace for other cases.
+function! sj#elm#JoinLines(text)
+  return substitute(substitute(a:text, '\s*\n\s*,', ',', 'g'), '\s*\n\s*', ' ', 'g')
+endfunction
+
+" function! sj#elm#SplitParts(from, to)
+"
+" Splits a section of a line according to bracing characters
+" rules.
+"
+" ex:
+"   "[1,2,3]"
+"   v
+"   [ "[ 1", ", 2", ", 3", "]" ]
+"
+"   "{a | foo = bar, baz = buzz}"
+"   v
+"   [ "{ a", "| foo = bar", ", baz = buzz", "}" ]
+function! sj#elm#SplitParts(from, to)
+  call sj#PushCursor()
+  let skip = sj#SkipSyntax(['elmString', 'elmTripleString', 'elmComment'])
```

Comment:
You could call the filetype-specific skip function you're prepared, I think it returns the same value.

Reply:
Ooops, forgot that one :sweat_smile: thanks.